buildSrc/src/main/java/com/google/firebase/gradle/plugins/GmavenVersionChecker.kt

```diff
@@ -0,0 +1,56 @@
+package com.google.firebase.gradle.plugins
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+
+abstract class GmavenVersionChecker : DefaultTask() {
+
+  @get:Input abstract val groupId: Property<String>
+
+  @get:Input abstract val artifactId: Property<String>
+
+  @get:Input abstract val latestReleasedVersion: Property<String>
+
+  @get:Input abstract val version: Property<String>
+
+  @TaskAction
+  fun run() {
+    val mavenHelper = GmavenHelper(groupId.get(), artifactId.get())
+    val latestMavenVersion = mavenHelper.getLatestReleasedVersion()
+    // Either the Maven metadata does not exist, or the library hasn't been released
+    if (latestMavenVersion.isEmpty()) {
+      return
+    }
+    if (version.get() == latestReleasedVersion.get()) {
+      throw GradleException(
+        "Current and latest released versions from gradle.properties are the same (${version.get()})" +
```

Comment:
Could you use the var names here? IMO it would make it easier to understand what we refer to by "current" and "latest".

Reply:
Here and below